src/obfs/auth_chain.c

```diff
@@ -615,8 +1109,8 @@ int auth_chain_a_client_udp_pre_encrypt(obfs *self, char **pplaindata, int datal
     char auth_data[3];
     uint8_t hash[16];
     ss_md5_hmac_with_key((char *) hash, auth_data, 3, server->key, server->key_len);
-    int rand_len = udp_get_rand_len(&local->random_client, hash);
-    VLA(uint8_t, rand_len, rnd_data);
+    int rand_len = auth_chain_udp_get_rand_len(&local->random_client, hash);
+    uint8_t *rnd_data = (uint8_t *)malloc(rand_len);
```

Comment:
The VLA macro may get benefit from GCC or clang. It's not a good idea replace it with malloc.